daemons/controld/controld_metadata.c

```diff
@@ -173,6 +173,32 @@ ra_param_from_xml(xmlNode *param_xml)
     return p;
 }
 
+// Check what version of the OCF standard a resource agent supports
+static void
+check_ra_ocf_version(struct ra_metadata_s *md, const char *key,
+                     xmlNode *version_element)
+{
+    xmlChar *content = NULL;
+
+    if (version_element != NULL) {
+        content = xmlNodeGetContent(version_element);
+    }
+    if (content == NULL) {
+        crm_warn("%s does not advertise OCF version supported", key);
+        return;
+    }
+
+    if (compare_version((const char *) content, "2") >= 0) {
+        crm_warn("%s supports OCF version %s and we don't (agent may not work "
+                 "properly with this version of Pacemaker)",
+                 key, (const char *) content);
+    } else {
```

Comment:
That's how we defined it: if an agent can't cope with treated as described in 1.1 it has to advertise 2.x at least - right?
Well - usual minor version definition ...
Would it still make sense to give a statement here that make it obvious like
"pacemaker supporting OCF 1.1 may result in not all features of %s supporting OCF %s being used".

Reply:
That's actually true within the 1.x series -- Pacemaker <2.1.0 supporting OCF 1.0 won't use reload-agent, etc., and Pacemaker 2.1.0 wouldn't use any features added in 1.2 and later (if any). That's somewhat expected, and I don't think it's particularly worth mentioning. If there's ever an OCF 2, it would mean a backward-incompatible break, so even existing features might not work as intended.

Reply:
I didn't suggest to change warnings for OCF >=2.
And of course if an agent supports OCF 1.3 it is the expected behavior that it will work treated as 1.1 by pacemaker but not everything might be as new and shiny as expected.
Thought a small heads-up (not necessarily a warning) might be interesting for the user that the resource-agent he just adopted might not work at its full potential with the pacemaker-version and that - whoever has made the resource-agent - might have done most of the testing in a different environment.

Reply:
Ah, I understand now. That could be useful, maybe at notice or info level.